plugins/out_datadog/datadog_conf.c

```diff
@@ -43,6 +43,46 @@ struct flb_out_datadog *flb_datadog_conf_create(struct flb_output_instance *ins,
     ctx->ins = ins;
     ctx->nb_additional_entries = 0;
 
+    tmp = flb_output_get_property("proxy", ins);
+    if (tmp) {
```

Comment:
what about using flb_uri_create() here ?

https://github.com/fluent/fluent-bit/blob/master/src/flb_uri.c#L45

Reply:
This logic of parsing the `proxy` parameter is inherited from the `out_http` plugin:
https://github.com/fluent/fluent-bit/blob/master/plugins/out_http/http_conf.c#L61

Maybe this code snippet should be extracted from `http_conf.c` in order to be reused by `datadog_conf.c` but I’m not sure `flb_uri_create()` would do the job since the main goal of this code snippet is to extract the proxy host and tcp port whereas `flb_uri_create()` is for splitting an uri by `/`-separated tokens.

Reply:
`flb_utils_split_url()` will do the job!